FluentUI/src/main/java/com/microsoft/fluentui/contextualcommandbar/ContextualCommandBar.kt

```diff
@@ -0,0 +1,276 @@
+                    itemPaddingVertical,
+                    itemPaddingHorizontal,
+                    itemPaddingVertical
+            )
+            setBackgroundColor(ThemeUtil.getColor(
+                    context,
+                    R.attr.fluentuiContextualCommandBarDismissBackgroundColor
+            ))
+            imageTintList = ColorStateList.valueOf(ThemeUtil.getColor(
+                    context,
+                    R.attr.fluentuiContextualCommandBarDismissIconTintColor
+            ))
+
+            setOnClickListener {
+                dismissListener?.invoke()
+            }
+        }
+        dismissButtonContainer!!.addView(dismissButton)
+        dismissButton!!.layoutParams.apply {
+            height = ViewGroup.LayoutParams.MATCH_PARENT
+            width = resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_button_width
+            )
+        }
+
+        bringChildToFront(dismissButtonContainer)
+    }
+
+    private fun setDismissButtonVisible(visible: Boolean) {
+        if (dismissIcon == 0) {
+            // Skip if there is no icon
+            return
+        }
+        if (visible && dismissButton == null) {
+            addDismissButton()
+        }
+
+        val paddingEnd = if (visible) {
+            resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_button_width
+            ) + resources.getDimensionPixelSize(
+                    R.dimen.fluentui_contextual_command_bar_dismiss_gap_width
+            )
+        } else 0
+        commandContainer.setPaddingRelative(0, 0, paddingEnd, 0)
+        dismissButtonContainer?.isVisible = visible
+    }
+
+    fun setItemGroups(itemGroups: List<CommandItemGroup>) {
+        commandContainer.removeAllViews()
+
+        for ((groupIdx, itemGroup) in itemGroups.withIndex()) {
+            val items = itemGroup.items
+            if (items.isEmpty()) {
+                continue
+            }
+
+            val groupContainer = LinearLayout(context)
+            commandContainer.addView(groupContainer)
+            if (groupIdx != itemGroups.size - 1) {
+                (groupContainer.layoutParams as LinearLayout.LayoutParams).apply {
+                    marginEnd = groupSpace
+                }
+            }
+
+            addItemViewsToGroup(items, groupContainer)
+        }
+    }
+
+    private fun addItemViewsToGroup(items: List<CommandItem>, groupContainer: ViewGroup) {
+        for ((idx, item) in items.withIndex()) {
+            val itemView = ImageView(context).apply {
+                setPaddingRelative(
+                        itemPaddingHorizontal,
+                        itemPaddingVertical,
+                        itemPaddingHorizontal,
+                        itemPaddingVertical
+                )
+                setImageResource(item.getIcon())
+                isSelected = item.isSelected() && item.isEnabled()
```

Comment:
is this assumption correct? for example if you have selected text which is all Bold, the Bold command should be selected, but if the content is read only it may also be disabled? read-only content should generally not disable selection, but I am not sure if the design of the CCB UX is that it should reflect context changes in this case...